guide/src/class.md

```diff
@@ -15,57 +15,111 @@ struct MyClass {
 }
 ```
 
-The above example generates implementations for `PyTypeInfo` and `PyTypeObject` for `MyClass`.
+The above example generates implementations for `PyTypeInfo`, `PyTypeObject`
+and `PyClass` for `MyClass`.
 
-## Get Python objects from `pyclass`
+Specifically, the following implementation is generated.
+
+```rust
+use pyo3::prelude::*;
+use pyo3::{PyClassShell, PyTypeInfo};
+
+struct MyClass {
+    num: i32,
+    debug: bool,
+}
+
+impl pyo3::pyclass::PyClassAlloc for MyClass {}
+
+impl PyTypeInfo for MyClass {
+    type Type = MyClass;
+    type BaseType = pyo3::types::PyAny;
+    type ConcreteLayout = PyClassShell<Self>;
+
+    const NAME: &'static str = "MyClass";
+    const MODULE: Option<&'static str> = None;
+    const DESCRIPTION: &'static str = "This is a demo class";
+    const FLAGS: usize = 0;
+
+    #[inline]
+    unsafe fn type_object() -> &'static mut pyo3::ffi::PyTypeObject {
+        static mut TYPE_OBJECT: pyo3::ffi::PyTypeObject = pyo3::ffi::PyTypeObject_INIT;
+        &mut TYPE_OBJECT
+    }
```

Comment:
This should be:
```rust
    unsafe fn type_object() -> *mut pyo3::ffi::PyTypeObject {
        static TYPE_OBJECT: UnsafeCell<pyo3::ffi::PyTypeObject> = UnsafeCell::new(pyo3::ffi::PyTypeObject_INIT);
        TYPE_OBJECT.get()
    }
```

since mutable references to `static mut` are nearly always undefined behavior.

See https://users.rust-lang.org/t/why-is-static-mut-bad/32888
and https://github.com/rust-lang/rust/issues/53639

The function still needs to be `unsafe` so users can't safely mis-implement it.

Reply:
Yeah, but this is not a new part that this PR introduced.
It is copy-pasted from type_object.rs.

Though I completely agree that it is problematic, maybe what I need is a mechanism that inhibits access to uninitialized TYPE_OBJECT. 
Let's do it in another PR.

Reply:
Continued at #699